components/Settings.js

```diff
@@ -203,33 +204,109 @@ const MenuButton = React.memo(({ name, select, selected }) => {
   )
 })
 
-const Presets = React.memo(({ show, selected, presets, toggle, remove, apply, undo }) => {
+const Preset = React.memo(({ remove, apply, index, selected, preset }) => (
+  <div className="preset-container">
+    <button
+      className="preset-tile"
+      style={{
+        cursor: index === selected ? 'initial' : 'pointer',
+        backgroundImage: `url('${preset.icon}'`,
+        backgroundColor: preset.icon ? 'initial' : preset.backgroundColor,
+        boxShadow: index === selected ? `inset 0px 0px 0px 2px ${COLORS.SECONDARY}` : 'initial'
+      }}
+      onClick={() => apply(index, preset)}
+    />
+    {preset.custom ? (
+      <button className="preset-remove" onClick={() => remove(index)}>
+        <Remove />
+      </button>
+    ) : null}
+    <style jsx>
+      {`
+        button {
+          outline: none;
+          border: none;
+          background: transparent;
+          cursor: pointer;
+          padding: 0;
+        }
+
+        .preset-container {
+          display: flex;
+          position: relative;
+          flex: 0 0 96px;
+          height: 96px;
+          margin-right: 8px;
+        }
+
+        .preset-tile {
+          flex: 1;
+          border-radius: 3px;
+          background-repeat: no-repeat;
+          background-position: center center;
+          background-size: contain;
+        }
+
+        .preset-remove {
+          display: flex;
+          align-items: center;
+          justify-content: center;
+          position: absolute;
+          padding: 0;
+          top: 6px;
+          right: 6px;
+          width: 11px;
+          height: 11px;
+          border-radius: 999px;
+          background-color: ${COLORS.SECONDARY};
+        }
+      `}
+    </style>
+  </div>
+))
+
+const Presets = React.memo(({ show, create, toggle, undo, presets, selected, remove, apply }) => {
+  const customPresetsLength = presets.length - DEFAULT_PRESETS.length
+
   return (
     <div className="settings-presets">
       <div className="settings-presets-header">
         <span>Presets</span>
+        {show && (
+          <button
+            className="settings-presets-create"
+            onClick={create}
+            style={{ pointerEvents: selected === null ? 'initial' : 'none' }}
+          >
+            create +
+          </button>
+        )}
         <button className="settings-presets-arrow" onClick={toggle}>
           {show ? <Arrows.Up /> : <Arrows.Down />}
         </button>
       </div>
       {show ? (
         <div className="settings-presets-content">
-          {presets.map((preset, i) => (
-            <button
+          {presets.slice(0, customPresetsLength).map((preset, i) => (
```

Comment:
I felt this way was more understandable than doing a single loop and figuring out where to insert the divider. I don't like how dependent this code is on array indexes, but I'm not sure using id's and then having to sort them is any better.

Reply:
Again let's use `filter(p => p.custom)` and `filter(p => !p.custom)` here instead of `slice` 👍